domain_models/fields.py

```diff
@@ -185,9 +196,17 @@ def __init__(self, related_model_cls, default=None, required=False):
         self.related_model_cls = related_model_cls
 
     def _converter(self, value):
-        """Convert raw input value of the field."""
+        """Convert raw input value of the field.
+
+        :param object value:
+        :rtype object:
+        """
         if type(value) is not self.related_model_cls.Collection:
-            value = self.related_model_cls.Collection(value)
+            value = self.related_model_cls.Collection([
```

Comment:
Btw, we can make small utility function in current module:

```python
def _get_model_instance(model_cls, data):
    return model_cls(**data) if isinstance(data, dict) else data

# And after it could look like:
value = self.related_model_cls.Collection([_get_model_instance(self.related, item) for item in value])
```